zerver/lib/cache.py

```diff
@@ -395,7 +395,7 @@ def changed(fields: List[str]) -> bool:
     if changed(['is_guest']):
         cache_delete(active_non_guest_user_ids_cache_key(user_profile.realm_id))
 
-    if changed(['email', 'full_name', 'short_name', 'id', 'is_mirror_dummy']):
+    if changed(['email', 'full_name', 'short_name', 'id', 'is_mirror_dummy', 'delivery_email']):
```

Comment:
We don't need this, because the data doesn't appear in the `display_recipient` cache.